File: src/main/java/land/leets/domain/comment/usecase/GetCommentsImpl.java
package land.leets.domain.comment.usecase;

import land.leets.domain.comment.domain.Comment;
import land.leets.domain.comment.domain.repository.CommentRepository;
import land.leets.domain.comment.presentation.dto.CommentResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class GetCommentsImpl implements GetComments {

    private final CommentRepository commentRepository;

    @Override
    public List<CommentResponse> execute(Long applicationId) {
        List<Comment> comments = commentRepository.findAllByApplicationId(applicationId);
        return comments.stream()
                .map(CommentResponse::from)
                .toList();
    }
}

File: src/main/kotlin/land/leets/domain/comment/usecase/GetCommentsImpl.kt
package land.leets.domain.comment.usecase

import land.leets.domain.comment.domain.repository.CommentRepository
import land.leets.domain.comment.presentation.dto.CommentResponse
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class GetCommentsImpl(
    private val commentRepository: CommentRepository
) : GetComments {
    @Transactional(readOnly = true)
    override fun execute(applicationId: Long): List<CommentResponse> {
        val comments = commentRepository.findAllByApplicationId(applicationId)
        return comments.map { CommentResponse.from(it) }
    }
}
